src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -228,6 +232,18 @@ private String configureBucket() {
     return bucket;
   }
 
+  private void configureTable() {
+    AbstractBigQuerySinkConfig config = getConfig();
+    if (!(config instanceof BigQuerySinkConfig)) {
```

Comment:
changing behavior based on the type of config is not a good pattern. Instead, this can be a protected method that only the sink overrides.

Reply:
I'm also not sure I understand why this is required. Is the sink not already setting these properties? How is it currently working if it is not?

Reply:
Done.

Reply:
No this property not setting. Current property need for set clustering configuration only when new table will be created.
Example: destination table exists and not support clustering, in this case if clustering propertyes be set, system will get exeption during connection to BQ table.